### PR TITLE
Use opts.path_display to customize

### PR DESCRIPTION
**Pull request by delphinus**
Some telescope.nvim's builtin pickers support `path_display` option to customize its way to show filenames. This PR brings it here.

See [:h telescope.defaults.path_display](https://github.com/nvim-telescope/telescope.nvim/blob/bd9e8a7eac1b3f921d293866abf7574d49cb610c/doc/telescope.txt#L261).

When you do not set `path_display`, it uses the same way as previous one.

```lua
require("telescope").extensions.frecency.frecency {
  path_display = { "shorten", "absolute" },
  -- or you can set func to edit
  path_display = function(_, filename)
    -- edit here
    return filename
  end,
}
```

### CODE REVIEW

File: lua/telescope/_extensions/frecency.lua [hunk 1]
```diff
@@ -35,22 +35,16 @@ local state = {
...
-    -- check relative to home/current
-    if vim.startswith(filename, os_home) then
-      filename = "~/" .. Path:new(filename):make_relative(os_home)
-    elseif filename ~= original_filename then
-      filename = "./" .. filename
-    end
+local function filepath_formatter(opts)
+  local path_opts = {}
+  for k, v in pairs(opts) do
+    path_opts[k] = v
```
Comment (Conni2461) [inline lua/telescope/_extensions/frecency.lua:41]: whats the point of this loop.

we just copy opts into path_opts arn't we?

Comment (Conni2461) [reply lua/telescope/_extensions/frecency.lua:41]: yeah we can do it like this

**Comment by Conni2461** [comment]
thanks :)

File: lua/telescope/_extensions/frecency.lua (hunk 1 above)
Comment (delphinus) [reply lua/telescope/_extensions/frecency.lua:41]: Yes, we need to clone opts here to overwrite later if you use tags to filter.

**Pull request merged**